Slim/Control/Queries.pm

```diff
@@ -4734,24 +4739,48 @@ sub worksQuery {
 		$sql .= ' ';
 	}
 
-	$sql .= " GROUP BY $groupBy ";
-
 	my $dbh = Slim::Schema->dbh;
 
+	my $stillScanning = Slim::Music::Import->stillScanning();
+
 	# Get count of unique composers, the count is cached until the next rescan done event
-	my $cacheKey = md5_hex($sql . join( '', @{$p} ) . 'composerCount' . Slim::Music::VirtualLibraries->getLibraryIdForClient($client));
+	my $ccSql = sprintf($sql . " GROUP BY composer.id", "'c'");
+	my $cacheKey = md5_hex($ccSql . join( '', @{$p} ) . Slim::Music::VirtualLibraries->getLibraryIdForClient($client));
 
 	my $composerCount = $cache->{$cacheKey};
 	if ( !$composerCount ) {
-		my $ccSql = sprintf($sql, "composer.id AS cid");
 		my $total_sth = $dbh->prepare_cached( qq{
-			SELECT COUNT(DISTINCT cid)  FROM ( $ccSql ) AS t1
+			SELECT COUNT(1)  FROM ( $ccSql ) AS t1
 		} );
 
 		$total_sth->execute( @{$p} );
 		($composerCount) = $total_sth->fetchrow_array();
 		$total_sth->finish;
 	}
+	if ( !$stillScanning ) {
+		$cache->{$cacheKey} = $composerCount;
+	}
```

Comment:
Please move this inside the condition, as only there we would define a new value. Otherwise you're just storing in the cache what you just read from it.